Mathlib/Analysis/InnerProductSpace/Laplace.lean

```diff
@@ -0,0 +1,181 @@
+the canonical covariant tensor of `E`, as defined and discussed in
+`Mathlib.Analysis.InnerProductSpace.CanonicalTensor`.
+
+We show that the Laplace operator is `ℂ`-linear on continuously differentiable functions, and
+establish the standard formula for computing the Laplace operator in terms of orthonormal bases of
+`E`.
+-/
+
+open InnerProductSpace TensorProduct Topology
+
+section secondDerivativeAPI
+
+/-!
+## Supporting API
+
+The definition of the Laplace Operator of a function `f : E → F` involves the notion of the second
+derivative, which can be seen as a continous multilinear map `ContinuousMultilinearMap 𝕜 (fun (i :
+Fin 2) ↦ E) F`, a bilinear map `E →ₗ[𝕜] E →ₗ[𝕜] F`, or a linear map on tensors
+`E ⊗[𝕜] E →ₗ[𝕜] F`. This section provides convenience API to convert between these notions.
+-/
+
+variable
+  {𝕜 : Type*} [NontriviallyNormedField 𝕜]
+  {E : Type*} [NormedAddCommGroup E] [NormedSpace 𝕜 E]
+  {F : Type*} [NormedAddCommGroup F] [NormedSpace 𝕜 F]
+  {G : Type*} [NormedAddCommGroup G] [NormedSpace 𝕜 G]
+
+variable (𝕜) in
+/--
+Convenience reformulation of the second iterated derivative, as a map from `E` to bilinear maps
+`E →ₗ[ℝ] E →ₗ[ℝ] ℝ
+-/
+noncomputable def bilinear_of_iteratedFDeriv_two (f : E → F) : E → E →ₗ[𝕜] E →ₗ[𝕜] F :=
+  fun x ↦ (fderiv 𝕜 (fderiv 𝕜 f) x).toLinearMap₂
+
+/--
+Expression of `bilinear_of_iteratedFDeriv_two` in terms of `iteratedFDeriv`.
+-/
+lemma bilinear_of_iteratedFDeriv_two_eq_iteratedFDeriv (f : E → F) (e e₁ e₂ : E) :
+    bilinear_of_iteratedFDeriv_two 𝕜 f e e₁ e₂ = iteratedFDeriv 𝕜 2 f e ![e₁, e₂] := by
+  simp [iteratedFDeriv_two_apply f e ![e₁, e₂], bilinear_of_iteratedFDeriv_two]
+
+variable (𝕜) in
+/--
+Convenience reformulation of the second iterated derivative, as a map from `E` to linear maps
+`E ⊗[𝕜] E →ₗ[𝕜] F`.
+-/
+noncomputable def tensor_of_iteratedFDeriv_two (f : E → F) : E → E ⊗[𝕜] E →ₗ[𝕜] F :=
+  fun e ↦ lift (bilinear_of_iteratedFDeriv_two 𝕜 f e)
+
+/--
+Expression of `tensor_of_iteratedFDeriv_two` in terms of `iteratedFDeriv`.
+-/
+lemma tensor_of_iteratedFDeriv_two_eq_iteratedFDeriv (f : E → F) (e e₁ e₂ : E) :
+    tensor_of_iteratedFDeriv_two 𝕜 f e (e₁ ⊗ₜ[𝕜] e₂) = iteratedFDeriv 𝕜 2 f e ![e₁, e₂] := by
+  rw [← bilinear_of_iteratedFDeriv_two_eq_iteratedFDeriv, tensor_of_iteratedFDeriv_two]
+  rfl
+
+end secondDerivativeAPI
+
+/-!
+## Definition of the Laplace Operator
+-/
+
+variable
+  {E : Type*} [NormedAddCommGroup E] [InnerProductSpace ℝ E] [FiniteDimensional ℝ E]
+  {F : Type*} [NormedAddCommGroup F] [NormedSpace ℝ F]
+  {G : Type*} [NormedAddCommGroup G] [NormedSpace ℝ G]
+  {f f₁ f₂ : E → F} {x : E}
+
+variable (f) in
+/--
+Definition of the Laplace operator for functions on real inner product spaces.
+-/
+noncomputable def Real.Laplace : E → F :=
+  fun x ↦ tensor_of_iteratedFDeriv_two ℝ f x (InnerProductSpace.canonicalCovariantTensor E)
+
+/--
+Introduce `Δ` as a notation for the Laplace operator.
+-/
+notation "Δ" => Real.Laplace
```

Comment:
Can you scope this notation? Other areas of mathematics (may or do) also use this.